### PR TITLE
Add a troubleshooting guide for serverless workflows

### PR DESCRIPTION
**Pull request by rgolangh**
This troubleshooting guide contains also the desciption of the HTTPS error you get when failing to work with a valid certificate.

Signed-off-by: Roy Golan <rgolan@redhat.com>

Fixes: https://issues.redhat.com/browse/FLPATH-1535

### CODE REVIEW

**Comment by rgolangh** [comment]
/hold

I want to add an exaple error for the https error

**Comment by rgolangh** [comment]
image
image
image

File: content/docs/serverless-workflows/troubleshooting.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+## Workflow Errors
+
+### Problem: Workflow execution fails
+
+**Solution:**
+
+1. Examine the container log of the workflow
+    ```console
+        kubectl logs my-workflow-xy73lj
```
Comment (masayag) [inline content/docs/serverless-workflows/troubleshooting.md:32]: Do you think we should provide info on how to get from the Sonataflow CR to the pod itself?

File: content/docs/serverless-workflows/troubleshooting.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    ```
+
+### Problem: Workflow is not listed by the orchestrator plugin
+
+**Solution:**
+
+1. Examine the container log of the workflow
```
Comment (masayag) [inline content/docs/serverless-workflows/troubleshooting.md:39]: First, we should check for the pod status which might not be started.
Perhaps the first step should be checking the sonataflow status?
Then checking the status of the pod what might not be started and therefore no logs.

File: content/docs/serverless-workflows/troubleshooting.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    ```console
+        kubectl logs my-workflow-xy73lj
+    ```
+
+2. Most probably the Data index service was unready when the workflow started.
```
Comment (masayag) [inline content/docs/serverless-workflows/troubleshooting.md:44]: Should we include a sentence such as: *Workflows are being listed from the Data-Index service. Each workflow registers itself to the Data-Index service.*
Just to provide more context to this bullter.

File: content/docs/serverless-workflows/troubleshooting.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+   Typically this is what the log shows:
+    ```console
+        2024-07-24 21:10:20,837 ERROR [org.kie.kog.eve.pro.ReactiveMessagingEventPublisher] (main) Error while creating event to topic kogito-processdefinitions-events for event ProcessDefinitionDataEvent {specVersion=1.0, id='586e5273-33b9-4e90-8df6-76b972575b57', source=http://mtaanalysis.default/MTAAnalysis, type='ProcessDefinitionEvent', time=2024-07-24T21:10:20.658694165Z, subject='null', dataContentType='application/json', dataSchema=null, data=org.kie.kogito.event.process.ProcessDefinitionEventBody@7de147e9, kogitoProcessInstanceId='null', kogitoRootProcessInstanceId='null', kogitoProcessId='MTAAnalysis', kogitoRootProcessId='null', kogitoAddons='null', kogitoIdentity='null', extensionAttributes={kogitoprocid=MTAAnalysis}}: java.util.concurrent.CompletionException: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: sonataflow-platform-data-index-service.default/10.96.15.153:80 
+    ```
+
+3. Make sure the Data Index is ready, and restart the workflow
+    ```console
+        $ oc get pods sonataflow-platform-data-index-service-546f59f89f-b7548
```
Comment (masayag) [inline content/docs/serverless-workflows/troubleshooting.md:52]: let's maintain consistency by the cli used in this guide (either `oc` or `kubectl`, but only one of them)

we should also specify the namespace of the platform. If the orchestrator is installed by the OCP chart or the operator, then the destination should be the `sonataflow-infra` namespace.

I'm not concerned about advanced users that creates their own data-index by adding platform per-namespace explicitly.

**Review by masayag — COMMENTED** [review]
Thank you!

File: content/docs/serverless-workflows/troubleshooting.md (hunk 1 above)
Comment (rgolangh) [reply content/docs/serverless-workflows/troubleshooting.md:32]: I think not necessarily, but I created a section or error types, and there a deployment error is where examining the SF will be handy

File: content/docs/serverless-workflows/troubleshooting.md (hunk 4 above)
Comment (rgolangh) [reply content/docs/serverless-workflows/troubleshooting.md:52]: good catch, in orchestrator-k8s it is per ns

File: content/docs/serverless-workflows/troubleshooting.md (hunk 2 above)
Comment (rgolangh) [reply content/docs/serverless-workflows/troubleshooting.md:39]: This is when the workflow executed but failed. 
There should be another entry for when the workflow immediatly fails in the UI

File: content/docs/serverless-workflows/troubleshooting.md (hunk 3 above)
Comment (masayag) [reply content/docs/serverless-workflows/troubleshooting.md:45]: what about this comment? needed or not?

**Comment by rgolangh** [comment]
yes

**Review by masayag — APPROVED** [review]

**Pull request merged**